src/Entities/Player.cpp

```diff
@@ -2063,8 +2082,26 @@ bool cPlayer::LoadFromDisk(cWorldPtr & a_World)
 		return true;
 	}
 
-	// Load from the offline UUID file, if allowed:
+	// Check for old offline UUID filename, if it exists migrate to new filename
 	cUUID OfflineUUID = cClientHandle::GenerateOfflineUUID(GetName());
+	auto OldFilename = GetUUIDFileName(GetOldStyleOfflineUUID(GetName()));
+	LOG(OldFilename.c_str());
+	auto NewFilename = GetUUIDFileName(m_UUID);
+	// Only move if there isn't already a new file
+	if (!cFile::IsFile(NewFilename) && cFile::IsFile(OldFilename))
+	{
+		cFile::CreateFolder(NewFilename.substr(0, 10));  // Ensure folder exists to move to
```

Comment:
I think we've come to the point where abstracting the folder name away into a separate function is necessary, instead of the magic substrings.

Please add a function `static AString GetUUIDFolderName(const cUUID & a_Uuid)` that returns the folder (relative to current folder) in which the player file for the specified UUID is located, then use `cFile::CreateFolderRecursive()` on it here and in `cPlayer::SaveToDisk()`.

Reply:
Done